### PR TITLE
[Woo POS] Custom card component to remove simulation of the light source from the top right

### LINKED ISSUE #12492: [Woo POS] Custom card component to remove simulation of the light source from the top right

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12492

### Description

The PR implements custom card components that draw a shadow for elevation without taking into account light source.

Also the PR adds shadow to the bottom of the cart list

Discussion - p1725009492618169-slack-C070SJRA8DP

### Testing information

* Use the POS and notice that the shadows that illustrates elevation on the cards are drawn below the elements.
* Notice that there is a shadow appears when cart is scrollable but not at the bottom
* Put extra attention on potential performance degradation - scrolling of the lists

### The tests that have been performed

I looked into what I mentioned into the testing information

### Images/gif

https://github.com/user-attachments/assets/d65f00c8-5cf1-4da0-92ce-29cf2f5c18bc

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosCard.kt [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.isTraversalGroup
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun WooPosCard(
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosCard.kt:44]: ❓ What do you think about adding a comment to this class that explains the gist of why custom card is used and how it affects shadows?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov. I've left one minor comment that is non-blocking for the merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosCard.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosCard.kt:44]: Good call. Added it here

93a0a0584e7758d23e13d8970a7fada20e916040

**Pull request merged**